src/rgw/rgw_op.cc

```
@@ -590,7 +590,14 @@ void rgw_add_to_iam_environment(rgw::IAM::Environment& e, std::string&& key, std
 }
 
 void rgw_add_to_iam_environment(rgw::IAM::Environment& e, const std::string& key, const std::string& val){
-  e[key] = val;
+  if (!val.empty())
+    e[key] = val;
```

Quick question: is it an error condition if val is empty? An expected result? If so, instead of silent failure, should we either throw if it's an error, or return some indication if it's expected?

---

We should allow empty values. For s3:Prefix, for example, s3:Prefix being present and equal to the empty string is different from s3:Prefix being absent.